src/cli/mod.rs

```diff
@@ -0,0 +1,22 @@
+use crate::error::Error;
+use async_trait::async_trait;
+use clap::Args;
+
+/// The trait that represents the abstraction of a command line handler. To implement a new command
+/// line operation, implement this trait and register it, see `register::register_cli_command` for
+/// how to register.
+///
+/// Note that this trait does not support a stateful implementation as we assume CLI commands are all
+/// constructed from scratch. Initialize the states in the `handle` method.
+#[async_trait]
+pub trait CommandLineHandler {
+    /// The request to process.
+    ///
+    /// NOTE that this parameter is used to generate the command line arguments.
+    /// Currently we are directly integrating with `clap` crate. In the future we can use our own
+    /// implementation to abstract away external crates. But this should be good for now.
+    type Request: std::fmt::Debug + Args;
```

Comment:
Let's change `Request` to `Argument` so it is clear that the expected type is the argument for the cli command. Please, also update the comment to clarify that what we expect here is the set of commands supported by the cli command.

Reply:
Updated